operator/controllers/loki/lokistack_controller.go

```diff
@@ -142,14 +142,12 @@ func (r *LokiStackReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 func handleDegradedError(ctx context.Context, c client.Client, req ctrl.Request, err error) (ctrl.Result, error) {
 	var degraded *status.DegradedError
 	if errors.As(err, &degraded) {
-		err = status.SetDegradedCondition(ctx, c, req, degraded.Message, degraded.Reason)
-		if err != nil {
-			return ctrl.Result{}, err
+		derr := status.SetDegradedCondition(ctx, c, req, degraded.Message, degraded.Reason)
+		if derr != nil {
+			return ctrl.Result{}, derr
 		}
 
-		return ctrl.Result{
-			Requeue: degraded.Requeue,
-		}, nil
+		return ctrl.Result{Requeue: degraded.Requeue}, err
```

Comment:
I think this can be reverted to the state before. The new variable is not really necessary and the `return ctrl.Result{Requeue: degraded.Requeue}, nil` is intentional, so that the value of `degraded.Requeue` is actually used to decide whether the request should be retried or not.